assign-doi-datacite/src/main/java/no/unit/nva/doi/datacite/clients/DataCiteRestApiClient.java

```diff
@@ -0,0 +1,105 @@
+import java.net.http.HttpResponse;
+import java.time.Duration;
+import no.unit.nva.doi.datacite.clients.exception.ClientException;
+import no.unit.nva.doi.datacite.customerconfigs.CustomerConfig;
+import no.unit.nva.doi.datacite.customerconfigs.CustomerConfigException;
+import no.unit.nva.doi.datacite.customerconfigs.CustomerConfigExtractor;
+import no.unit.nva.doi.datacite.restclient.models.DoiStateDto;
+import no.unit.nva.doi.datacite.restclient.models.DraftDoiDto;
+import no.unit.nva.doi.models.Doi;
+import nva.commons.core.paths.UriWrapper;
+import org.apache.http.HttpStatus;
+
+public class DataCiteRestApiClient extends HttpSender {
+
+    private  static final int TIMEOUT = 2000;
+
+    public static final String ACCEPT = "Accept";
+    public static final String JSON_API_CONTENT_TYPE = "application/vnd.api+json";
+    public static final String CONTENT_TYPE = "Content-Type";
+
+    private static final String AUTHORIZATION_HEADER = "Authorization";
+    public static final String DOIS_PATH_PARAMETER = "dois";
+    private final URI dataciteRestApiURI;
+    private final CustomerConfigExtractor customerConfigExtractor;
+
+    private final String doiHost;
+
+    public DataCiteRestApiClient(URI dataciteRestApiURI,
+                                 String doiHost,
+                                 CustomerConfigExtractor customerConfigExtractor,
+                                 HttpClient httpClient) {
+        super(httpClient);
+        this.dataciteRestApiURI = dataciteRestApiURI;
+        this.customerConfigExtractor = customerConfigExtractor;
+        this.doiHost = doiHost;
+    }
+
+    public Doi createDoi(URI customerId) throws ClientException {
+        var customer = customerConfigExtractor.getCustomerConfig(customerId);
+        var request = createPostDoiRequest(customer);
+        var response = sendRequest(request, HttpStatus.SC_CREATED);
+        return convertResponseToDoi(response);
+    }
+
+    public DoiStateDto getDoi(URI customerId, Doi doi) throws ClientException {
+        var customer = customerConfigExtractor.getCustomerConfig(customerId);
+        var request = createGetDoiRequest(customer, doi);
+        var response = sendRequest(request, HttpStatus.SC_OK);
+        return DoiStateDto.fromJson(response.body());
+    }
+
+    private Doi convertResponseToDoi(HttpResponse<String> response) {
+        DraftDoiDto responseBody = DraftDoiDto.fromJson(response.body());
+        return responseBody.toDoi().changeHost(doiHost);
+    }
+
+    private HttpRequest createGetDoiRequest(CustomerConfig customer, Doi doi)
+        throws CustomerConfigException {
+        return HttpRequest.newBuilder()
+                   .uri(requestTargetUriToDoi(doi))
+                   .GET()
+                   .header(ACCEPT, JSON_API_CONTENT_TYPE)
+                   .timeout(Duration.ofMillis(TIMEOUT))
+                   .headers(AUTHORIZATION_HEADER, customer.extractBasicAuthenticationString())
```

Comment:
Do we need to set auth header manually on a GET request? I experienced some problems myself when i used an authenticated http client on a GET request so i had to remove that and had to set header manually myself. is it some java 17 issue?

Reply:
so when both method and manualk header were set at the same time, it caused an error? it was exactly what happened on profile picture lookup as well. removing the method seemed to fix it

Reply:
And to answer the original question: yes, we need to set the auth header for the get request. (if it is a draft DOI, and not sending auth headers you get a 404 response)